### PR TITLE
Add GSoC information pages

### LINKED ISSUE #867: Foster Google Summer of Code (GSOC) 2022 for Knative
Steering to work with the community to promote, encourage, support GSOC 2022 for Knative

/assign

### PR DESCRIPTION
**Pull request by aliok**
Fixes #867

# Changes

- :gift: Add GSoC information pages. Only information that's necessary at this stage.
- Created `google-summer-of-code` directory
- Created 2 files.
- README.md contains generic information about GSoC independent from the year.
- gsoc-2022.md contains specific information about GSOC 22 at Knative.

/kind <kind>

### CODE REVIEW

**Comment by aliok** [comment]
cc @csantanapr @evankanderson

**Comment by aliok** [comment]
cc @matzew

File: google-summer-of-code/gsoc-2022.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  in the program this year.
+* Try Knative by going through the [getting started](https://knative.dev/docs/getting-started/) page.
+* Interact with the community using the [communication channels](#communication).
+* Browse the [idea list](#idea-list) below. The earlier, the better.
+* Check out the Knative community [website](https://knative.dev/docs/community/).
+
+Tips:
+
+* Ask for help if you need assistance. The community will select the contributors who are not only good technically.
+  Community expects the contributors to have interaction.
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:32]:
```suggestion
  The community expects the contributors to have interaction.
```

File: google-summer-of-code/gsoc-2022.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+* The preferred way of asking for assistance is using the [communication channels](#communication). In specific cases,
+  you can privately contact the mentor if you feel uncomfortable about the public communication.
+* Start exploring early!
+* Have good interaction, so that the community recognizes your name! Start by introducing yourself to the community.
+* Be polite!
+* Help others! Help the like-minded potential contributors in the public channels.
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:38]:
```suggestion
* Help others! Help like-minded potential contributors in the public channels.
```

File: google-summer-of-code/gsoc-2022.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+### Coding period
+
+To be added later
+
+## Mentors
+
+### Application period
+
+Any Knative community member can submit ideas and volunteer to mentor students. However, Google gives limited number of
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:50]:
```suggestion
Any Knative community member can submit ideas and volunteer to mentor students. However, Google gives a limited number of
```

File: google-summer-of-code/gsoc-2022.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+slots, so, community will have to select the best idea+contributor+mentor combination.
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:51]:
```suggestion
slots, so, the community will have to select the best idea+contributor+mentor combination.
```

File: google-summer-of-code/gsoc-2022.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  in the program this year, if you have experience with GSoC before.
+* Add your idea to the [idea list](#idea-list) below. The earlier, the better.
+* Monitor the [communication channels](#communication) for potential contributor questions.
+
+Expectations from mentors:
+
+* Add your ideas [below](#idea-list).
+* Make sure you have the capacity to mentor a contributor. It is not the entire Knative community mentoring a
+  contributor; it is YOU mentoring the contributor. This depends on the complexity of the project and the technical
+  knowledge of the contributor, but as a general rule, expect spending 4 hours a week.
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:71]:
```suggestion
  knowledge of the contributor, but expect to spend 4 hours a week as a general rule.
```

File: google-summer-of-code/gsoc-2022.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+* Help the contributor to not only finish their projects but also teach about matters that are part of an open source
+  community such as how project governance is, how communication in the community works and similar.
+* Read the [mentor guide](https://google.github.io/gsocguides/mentor/) of the program.
+* Do NOT share any private information about the contributor candidates publicly, such as if you would like to mentor a
+  specific candidate or if you think candidate is not good enough for the program. Reach out to the administrators for
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:76]:
```suggestion
  specific candidate or if you think the candidate is not good enough for the program. Reach out to the administrators for
```

File: google-summer-of-code/gsoc-2022.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+  that.
+
+Tips:
+
+* Be polite and understanding! Not everyone has your knowledge on the technology we are working on at Knative community.
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:81]:
```suggestion
* Be polite and understanding! Not everyone has your knowledge of the technology we are working on at Knative community.
```

File: google-summer-of-code/gsoc-2022.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+* Be professional! Don't forget that the contributors could be part of the community one day.
+* Send a link of this page to the contributors when you think they can benefit the tips written here.
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:83]:
```suggestion
* Send a link to this page to the contributors when you think they can benefit from the tips written here.
```

File: google-summer-of-code/gsoc-2022.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+* Do not hesitate to contact the [administrators](#organization-administrators).
+
+### Coding period
+
+To be added later
+
+## Communication
+
+Contributors are welcome to participate any talks and discussions in the Knative community channels.
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:92]:
```suggestion
Contributors are welcome to participate in any talks and discussions in the Knative community channels.
```

File: google-summer-of-code/gsoc-2022.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+
+* Knative users group: https://groups.google.com/g/knative-users
+    * Preferred communication channel. Introduce yourself here as a candidate contributor.
+* Slack: https://slack.knative.dev/ Explore the channels yourselves, however make sure you're checking these:
```
Comment (csantanapr) [inline google-summer-of-code/gsoc-2022.md:96]:
```suggestion
* Slack: https://slack.knative.dev/ Explore the channels yourselves, however, make sure you're checking these:
```

**Review by csantanapr — CHANGES_REQUESTED** [review]
Thanks @aliok this are great files to get started, just left comments for minor nits

**Comment by aliok** [comment]
@csantanapr Thanks for the review and the suggestions. I committed them all.

Mind having a look again?

**Pull request merged**